llvm/lib/Target/AArch64/MachineSMEABIPass.cpp

```diff
@@ -918,6 +987,43 @@ bool MachineSMEABI::runOnMachineFunction(MachineFunction &MF) {
       getAnalysis<EdgeBundlesWrapperLegacy>().getEdgeBundles();
 
   FunctionInfo FnInfo = collectNeededZAStates(SMEFnAttrs);
+
+  if (OptLevel != CodeGenOptLevel::None) {
+    // Propagate desired states forwards then backwards. We propagate forwards
+    // first as this propagates desired states from inner to outer loops.
```

Comment:
Perhaps I'm getting this backwards, but if it would be propagating forwards first then I'd expect that it would first propagate states from the outer to the inner loop? (it would start at BB0, then BB1/BB3, then BB2)

Reply:
Okay, then I'd say that's worth clarifying in this paragraph or alternatively you can just remove most of this paragraph because the comments below the graph explain it better.

Reply:
I've updated this comment, let me know if it reads better now :+1: